src/material/dialog/dialog-container.ts

```diff
@@ -105,9 +105,9 @@ export abstract class _MatDialogContainerBase extends BasePortalOutlet {
     // Save the previously focused element. This element will be re-focused
     // when the dialog closes.
     this._capturePreviouslyFocusedElement();
-    // Move focus onto the dialog immediately in order to prevent the user
+    // Immediately blur the previously focused element in order to prevent the user
     // from accidentally opening multiple dialogs at the same time.
-    this._focusDialogContainer();
+    this._preventDuplicateDialogOpens();
```

Comment:
This will kick focus back to the `body` and it could cause the screen reader to start reading out the top-level landmarks until the animation finishes. I think a better way to approach this would be to remove the logic that moves focus immediately and instead prevent new dialogs from opening while a previous one is still animating.

Reply:
For a longer-animation, that could be an issue. I don't have one of those ready atm to verify with.

The `blur` does cause issues with CI on iOS, presumably since that platform hates to move the focus to anything but form elements. 

I'll take a look at how to prevent new dialogs from opening when a dialog is in the process of opening/animating.